cms/__init__.py

```diff
@@ -60,21 +60,28 @@
 LANG_CPP = "cpp"
 LANG_CS = "cs"
 LANG_PASCAL = "pas"
-LANG_PYTHON = "py"
+LANG_PYTHON2 = "py2"
+LANG_PYTHON3 = "py3"
 LANG_PHP = "php"
 LANG_JAVA = "java"
 
+# Python 3 py_compile outputs to __pycache__/filename.<PYTHON3_COMPILE_NAME>.pyc
+# Using a '*' in this place does not appear to work correctly.
+# Format seems standard as cpython-<MAJOR><MINOR> for cpython.
+PYTHON3_COMPILE_NAME = "cpython-%d%d" % (3, 4)
```

Comment:
This is pretty flaky; as you noted 34 is tied to the current version of python3.

Try something like

    import subprocess
    try:
        # Returns "Python M.m.p\n"
        "".join(subprocess.check_output(['python3', '-V']).strip().split()[1].split('.')[:2])
    except CalledProcessError:
        # rip python3 (maybe we don't care about this not existing since it implies PEBKAC)